### PR TITLE
Fix T5/mT5 tests for TensorFlow

### PR DESCRIPTION
**Pull request by Rocketknight1**
Some tests used `tf.reduce_sum()` on a per-token loss. Since we now return a scalar loss to enable XLA compilation and match PyTorch, the expected loss in these tests is wrong. If we change the tests to use `tf.reduce_mean()` and divide the expected value by the number of tokens, then the tests work with both the modern and legacy losses.

### CODE REVIEW

**Review by ydshieh — COMMENTED** [review]
LGTM, thanks for the fix! Could you try to run it locally, so we are sure we won't see it again on Slack 🙏

**Review by ydshieh — APPROVED** [review]

**Comment by Rocketknight1** [comment]
@ydshieh I ran them all locally and they passed!
